tests/Functional/Driver/PDO/MySQL/DeadlockTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Doctrine\DBAL\Tests\Functional\Driver\PDO\MySQL;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver as DriverInterface;
+use Doctrine\DBAL\Driver\PDO\MySQL\Driver;
+use Doctrine\DBAL\Exception\DeadlockException;
+use Doctrine\DBAL\Tests\Functional\Driver\AbstractDriverTest;
+use Doctrine\DBAL\Tests\TestUtil;
+use ReflectionObject;
+use Throwable;
+
+use function pcntl_fork;
+use function sleep;
+
+class DeadlockTest extends AbstractDriverTest
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if (TestUtil::isDriverOneOf('pdo_mysql')) {
+            return;
+        }
```

Comment:
The changes being made in the library code affect all supported drivers while the test only covers a specific one. What prevents testing the change with all drivers? Specifically, `mysqli`.

Reply:
Noted, I wIll add the mysqli, too.

For the other databases i'm not sure that actually they raise deadlock exceptions.
It should be straightforward to add all supported databases here? 
As far as I can tell this is a specific feature to mysql and it's flavours.

Reply:
> For the other databases i'm not sure that actually they raise deadlock exceptions.

If they don't, then the test shouldn't fail if the deadlock exception wasn't thrown. It should only test the the transaction nesting level is reset when it _is_ thrown.